packages/validator/docs/actions.md

```diff
@@ -0,0 +1,292 @@
+# Actions
+
+If any action is specified in template, object `output` will be returned when processing a specification. This object is at first empty. By inserting or copying parameters from the given specification, it can construct the desired specification.
+
+## Copy action
+
+`__copy` object located inside `__actions` array of some parameter will copy the value of parameter it is nested in into provided `__target`, which is evaluated as an absolute path.
+
+### Template
+
+```json
+{
+  "company": {},
+  "inventory": {
+    "__arrayItem": {
+      "name": {},
+      "quantity": {}
+    },
+    "__actions": [
+      {
+        "__copy": {
+          "__target": "[ 'backups', '[[ \\'/\\', \\'company\\' ]]' ]"
+        }
+      }
+    ]
+  }
+}
+```
+---
+### Input
+
+```json
+{
+  "company": "anon",
+  "inventory": [
+    {
+      "name": "item1",
+      "quantity": 10
+    },
+    {
+      "name": "item2",
+      "quantity": 3
+    }
+  ]
+}
+```
+### Expected output
+
+```json
+{
+  "valid": true,
+  "messages": [],
+  "output": {
+    "backups": {
+      "anon": [
+        {
+          "name": "item1",
+          "quantity": 10
+        },
+        {
+          "name": "item2",
+          "quantity": 3
+        }
+      ]
+    }
+  }
+}
+```
+---
+
+## Insert action
+
+Keyword `__insert` works similarly to `__copy`, except it doesn't copy a value of parameter, but inserts value provided in `__value`
+
+### Template
+
+```json
+{
+  "original": {},
+  "__actions": [
+    {
+      "__insert": {
+        "__target": "[ 'example1' ]",
+        "__value": "inserted"
+      }
+    },
+    {
+      "__insert": {
+        "__target": "[ 'example2' ]",
+        "__value": {
+          "obj": {
+            "value": "inserted"
+          }
+        }
+      }
+    }
+  ]
+}
+```
+---
+### Input
+
+```json
+{
+  "original": {}
+}
+```
+### Expected output
+
+```json
+{
+  "valid": true,
+  "messages": [],
+  "output": {
+    "example1": "inserted",
+    "example2": {
+      "obj": {
+        "value": "inserted"
+      }
+    }
+  }
+}
+```
+---
+
+## Arrays in `__target`
+
+Arrays in `__target` can be accessed as regular items in arrays with `[x]`, where `x` is index of the item that will be accessed. However, in arrays inside `__target` last item of the array can be accessed with `[_]` and new item can be inserted into array with `[]`.
```

Comment:
Because this part is quite difficult to comprehend, when Sasa was going through this it was causing confusion about what the last element of the array is as it is not the last item of resulting array rather then last item of intermediary that is currently build. Bringing there something that already has a unified meaning to certain degree would cause even more confusion, also I can't really imagine a scenario where anything else than `[-1]` would be used, so perhaps not giving the possibility to access other elements might be a way to avoid undesired behavior.